app/src/main/java/com/wizeline/heroes/viewmodels/IMarvelDetailsViewModel.kt

```diff
@@ -0,0 +1,56 @@
+package com.wizeline.heroes.viewmodels
+
+import android.util.Log
+import androidx.lifecycle.MutableLiveData
+import androidx.lifecycle.ViewModel
+import com.wizeline.heroes.interfaces.IMarvelFragmentViewModel
+import com.wizeline.heroes.models.Character
+import com.wizeline.heroes.models.ComicDetails
+import com.wizeline.heroes.models.MarvelCharacterDetailsViewState
+import com.wizeline.heroes.models.SeriesDetails
+import com.wizeline.heroes.usecases.MarvelCharacterDetailsUseCase
+import dagger.hilt.android.lifecycle.HiltViewModel
+import io.reactivex.disposables.CompositeDisposable
+import javax.inject.Inject
+
+@HiltViewModel
+class MarvelDetailsViewModel @Inject constructor(val characterDetailsUseCase: MarvelCharacterDetailsUseCase) :
+    ViewModel(), IMarvelFragmentViewModel {
+
+    private var _marvelDetailsViewState = MutableLiveData(MarvelCharacterDetailsViewState())
+    val marvelDetailsViewState get() = _marvelDetailsViewState
+
+    private val compositeDisposable = CompositeDisposable()
```

Comment:
Remember to call `compositeDisposable.clear()` on the ViewModels' `onCleared` method to ensure resources are cleaned up from memory when the ViewModel is destroyed.

Reply:
Done.